python/ray/data/tests/test_strict_mode.py

```diff
@@ -111,6 +111,12 @@ def f(x):
     assert isinstance(batch["id"], np.ndarray), batch
 
 
+def test_strict_require_batch_size(ray_start_regular_shared):
```

Comment:
let's also add unit test for all other APIs calling map_batches under the hood, e.g. https://github.com/ray-project/ray/pull/34588/files#r1171781083 .

Reply:
batch_size is no longer required for CPU `map_batches`